File: ip-messaging/rest/messages/list-messages/list-messages.5.x.rb
require 'http'
require 'twilio-ruby'

# Get your Account SID, Auth Token, Service SID and Channel SID from twilio.com/console
account_sid = '<ACCOUNT_SID>'
auth_token = '<AUTH_TOKEN>'
service_sid = '<SERVICE_SID>'
channel_sid = '<CHANNEL_SID>'
@client = Twilio::REST::Client.new(account_sid, auth_token)

# List the messages
service = @client.ip_messaging.v2.services(service_sid)
channel = service.channels(channel_sid) 

channel.messages.each do |message|
    puts "#{message.sid} #{message.body}"
end

File: ip-messaging/rest/messages/list-messages/list-messages.5.x.cs
// Download the twilio-csharp library from twilio.com/docs/libraries/csharp
using System;
using Twilio;
using Twilio.Rest.Chat.V1.Service.Channel;

class Example
{
    static void Main(string[] args)
    {
        // Find your Account SID and Auth Token at twilio.com/console
        const string accountSid = "accountSid";
        const string authToken = "authToken";
        const string serviceSid = "serviceSid";
        const string channelSid = "channelSid";

        TwilioClient.Init(accountSid, authToken);

        // List all messages
        var messages = MessageResource.Read(serviceSid, channelSid);

        foreach (var message in messages)
        {
            Console.WriteLine(message.DateCreated);
        }
    }
}
